### ISSUE TITLE
GLIBCXX_3.4.29' not found

### ISSUE THREAD

**Issue by mokilcde** [RocketChat/Rocket.Chat#31258]
Upgrading Rocket.Chat from 6.4.8 to 6.5 does not work:

```
Dec 16 11:24:16 jitsi.k-c13.org systemd[1]: Started The Rocket.Chat server.
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]: Some indexes for collection 'rocketchat_analytics' could not be created:
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2>
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]: Some indexes for collection 'rocketchat_credential_tokens' could not be created:
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2>
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]: Some indexes for collection 'rocketchat_subscription' could not be created:
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2>
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]: Some indexes for collection 'rocketchat_uploads' could not be created:
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2>
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]: Some indexes for collection 'rocketchat_message' could not be created:
Dec 16 11:24:21 jitsi.k-c13.org rocketchat[129354]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]: /srv/rocketchat/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.j>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:       throw error;
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:       ^
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]: Error: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /srv/rocketchat>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Object.Module._extensions..node (internal/modules/cjs/loader.js:1173:18)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.load (internal/modules/cjs/loader.js:979:32)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.Mp.load (/srv/rocketchat/programs/server/runtime.js:37:33)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Function.Module._load (internal/modules/cjs/loader.js:819:12)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (internal/modules/cjs/loader.js:1003:19)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at require (internal/modules/cjs/helpers.js:107:18)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Object.<anonymous> (/srv/rocketchat/programs/server/npm/node_modules/isolated-vm/isolated-vm.js:1:-44)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module._compile (internal/modules/cjs/loader.js:1114:14)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.Mp._compile (/srv/rocketchat/programs/server/runtime.js:77:23)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.load (internal/modules/cjs/loader.js:979:32)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.Mp.load (/srv/rocketchat/programs/server/runtime.js:37:33)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Function.Module._load (internal/modules/cjs/loader.js:819:12)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (internal/modules/cjs/loader.js:1003:19)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at require (internal/modules/cjs/helpers.js:107:18)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at npmRequire (/srv/rocketchat/programs/server/npm-require.js:111:12)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.useNode (packages/modules-runtime.js:751:18)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at module (/srv/rocketchat/programs/server/packages/modules.js:202897:8)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_mod>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at module (app/integrations/server/lib/isolated-vm/isolated-vm.ts:2:43)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_mod>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at module (app/integrations/server/lib/triggerHandler.js:1:1149)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_mod>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at server/services/meteor/service.ts:1:72
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at module (server/services/meteor/service.ts:285:2)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_mod>
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]:   code: 'ERR_DLOPEN_FAILED'
Dec 16 11:24:38 jitsi.k-c13.org rocketchat[129354]: }
```

I did the update always the same way the last few years:
```
curl -L https://releases.rocket.chat/latest/download -o /tmp/rocket.chat.tgz
tar -xzf /tmp/rocket.chat.tgz -C /tmp
cd /tmp/bundle/programs/server && npm install
cd /srv/
mv rocketchat/ rocketchat.old
mv /tmp/bundle/ /srv/rocketchat
systemctl restart rocketchat
```

- Version of Rocket.Chat Server: 6.5.0
- Operating System:  Debian 11
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: v14.21.3
- MongoDB Version: 5.0.23

**Comment by Avnosx** [comment]
Hello ! It seems that it throws an error "Error: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found
" so It seems that either the libstdc++ needs an update or try installing gcc-9

**Comment by mokilcde** [comment]
I installed it but no changes

```
Dec 16 14:18:00 jitsi.k-c13.org systemd[1]: Started The Rocket.Chat server.
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]: Some indexes for collection 'rocketchat_analytics' could not be created:
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2, unique: true, key: { room._id: 1, date: 1 }, name: "room._id_1_date_1", partialFilterExpression: { type: "rooms" } }, existing index: { v: 2, unique: true, key: { room._id: 1, date: 1 }, name: "room._id_1_date_1" }
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]: Some indexes for collection 'rocketchat_credential_tokens' could not be created:
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2, key: { expireAt: 1 }, name: "expireAt_1", sparse: true, expireAfterSeconds: 0 }, existing index: { v: 2, key: { expireAt: 1 }, name: "expireAt_1", sparse: 1, expireAfterSeconds: 0 }
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]: Some indexes for collection 'rocketchat_message' could not be created:
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2, key: { pinnedBy._id: 1 }, name: "pinnedBy._id_1", sparse: true }, existing index: { v: 2, key: { pinnedBy._id: 1 }, name: "pinnedBy._id_1", sparse: 1 }
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]: Some indexes for collection 'rocketchat_subscription' could not be created:
Dec 16 14:18:06 jitsi.k-c13.org rocketchat[143345]:         An equivalent index already exists with the same name but different options. Requested index: { v: 2, key: { desktopNotifications: 1 }, name: "desktopNotifications_1", sparse: true }, existing index: { v: 2, key: { desktopNotifications: 1 }, name: "desktopNotifications_1", sparse: 1 }
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]: /srv/rocketchat/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:218
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:       throw error;
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:       ^
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]: Error: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /srv/rocketchat/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Object.Module._extensions..node (internal/modules/cjs/loader.js:1173:18)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.load (internal/modules/cjs/loader.js:979:32)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.Mp.load (/srv/rocketchat/programs/server/runtime.js:37:33)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Function.Module._load (internal/modules/cjs/loader.js:819:12)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (internal/modules/cjs/loader.js:1003:19)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at require (internal/modules/cjs/helpers.js:107:18)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Object.<anonymous> (/srv/rocketchat/programs/server/npm/node_modules/isolated-vm/isolated-vm.js:1:-44)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module._compile (internal/modules/cjs/loader.js:1114:14)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.Mp._compile (/srv/rocketchat/programs/server/runtime.js:77:23)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.load (internal/modules/cjs/loader.js:979:32)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.Mp.load (/srv/rocketchat/programs/server/runtime.js:37:33)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Function.Module._load (internal/modules/cjs/loader.js:819:12)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (internal/modules/cjs/loader.js:1003:19)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at require (internal/modules/cjs/helpers.js:107:18)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at npmRequire (/srv/rocketchat/programs/server/npm-require.js:111:12)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.useNode (packages/modules-runtime.js:751:18)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at module (/srv/rocketchat/programs/server/packages/modules.js:202897:8)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_modules/@meteorjs/reify/lib/runtime/index.js:52:22)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at module (app/integrations/server/lib/isolated-vm/isolated-vm.ts:2:43)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_modules/@meteorjs/reify/lib/runtime/index.js:52:22)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at module (app/integrations/server/lib/triggerHandler.js:1:1149)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_modules/@meteorjs/reify/lib/runtime/index.js:52:22)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at server/services/meteor/service.ts:1:72
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at module (server/services/meteor/service.ts:285:2)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.require (packages/modules-runtime.js:238:14)
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:     at Module.moduleLink [as link] (/srv/rocketchat/programs/server/npm/node_modules/meteor/modules/node_modules/@meteorjs/reify/lib/runtime/index.js:52:22) {
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]:   code: 'ERR_DLOPEN_FAILED'
Dec 16 14:18:22 jitsi.k-c13.org rocketchat[143345]: }
```

I checked the dependencies for `isolated_vm.node`
```
ldd /tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node
/tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node)
/tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node)
/tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node)
/tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/bundle/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node)
	linux-vdso.so.1 (0x00007fff371fd000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9175c31000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9175aed000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9175ad3000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f91758ff000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9175f10000)
```
This is the file in 6.4.8:
```
ldd /srv/rocketchat.old/programs/server/npm/node_modules/isolated-vm/out/isolated_vm.node
	linux-vdso.so.1 (0x00007ffd463cc000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc67cd8c000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc67cc48000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc67cc2e000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fc67cc0c000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc67ca38000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc67d067000)
```

This cannot work and `isolated_vm.node` is a part of the tar archive. Why is this changed in 6.5?

**Comment by Gummikavalier** [comment]
Check https://github.com/RocketChat/Rocket.Chat/issues/30612

In short, if you are running as root, try creating a separate account for RC.

**Comment by mokilcde** [comment]
I fixed it.
The first step was to upgrade my Debian box from 11 to 12.
After that I have to remove the already installed nodejs from Debian and install the version 14.x directly from them with a long timeout because of node version 14.

This is really a nightmare.

**Comment by MartinSchoeler** [comment]
Since the issue seems to be resolved, I will be closing this, thanks!

**Comment by anhtukhtn** [comment]
> I fixed it. The first step was to upgrade my Debian box from 11 to 12. After that I have to remove the already installed nodejs from Debian and install the version 14.x directly from them with a long timeout because of node version 14.
> 
> This is really a nightmare.

thank you very much. I got the same error and fixed with ubuntu:22.04 (install node 14 directly)
